Issue 28: add schema validation to pipeline

wdduncan opened the pull request:
see issues #28  #23

dwinston reviewed (commented):
we're scheduled to meet soon, so we can clear things up then.

my general comment at this point is that I would like some statement of the goal, and some brief elaboration of strategy/tactics as appropriate, in the main conversation part of the PR, to guide my reading of the file diffs. This is because I'm not clear on what is being attempted here -- I guess our prior conversation has been booted from my working memory. sorry!

wdduncan commented:
@dwinston I added graph  for the emsl and gold validation.  
Since these graphs validate multiple data sets, I reverted using ops to get the collection names; e.g.:
```
@graph()
def gold():
    validate_mongo_collection(gold_study_set_collection_name())
    validate_mongo_collection(gold_omics_processing_set_collection_name())
    validate_mongo_collection(gold_biosample_set_collection_name())
```
Not sure how to pass multiple collection names using the config.  But we can revisit how to do it best.
